pkg/apiclient/workflow/workflow.proto

```diff
@@ -33,6 +33,10 @@ message WorkflowListRequest {
   k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
   // Fields to be included or excluded in the response. e.g. "items.spec,items.status.phase", "-items.status.nodes"
   string fields = 3;
+  // Return workflows having this name prefix
+  string namePrefix = 4;
+  // Return workflows with name containing this value
+  string namePattern = 5;
```

Comment:
I'm wondering if we should make the API match the UI actually; `name` as a string and `nameFilter` as an email that defaults to `"Exact"`

Right now, it's not clear that these are mutually exclusive. We could add that in the comments, but having it only allow one at a time feels much more explicit

Reply:
Yeah I like that idea, so we could have `nameFilter` as `Exact` | `Contains` | `Prefix`, and then `nameValue` as `string`

I'm not familiar with the use case, but does it make more sense to have `Exact` as default instead of `Contains` ?

Reply:
`Exact` as default would be backward compatible, since previously it was the only usage of `name`